app/models/host/base.rb

```diff
@@ -187,6 +187,10 @@ def primary_interface_type(parser)
       interface_class(identifier).to_s
     end
 
+    def set_reported_data(parser)
+      # implemented in Host::Managed
```

Comment:
Should this raise an error?
```suggestion
      raise NotImplementedError
```

Reply:
If we'd raise an error here, we would not unbreak foreman.
`set_reported_data` is called in `populate_fields_from_facts`.

We could simply change `populate_fields_from_facts` in `Host::Managed`:

```ruby
def populate_fields_from_facts
  super
  set_reported_data
end
```

Reply:
That makes a lot more sense to me. However, I'm wondering if Host::Base should use the concern rather than Host::Managed. Can't discovered hosts store facts as well?

Reply:
if we do it this way, every inheriting class needs to "disable" this by noop, if we default to noop, that's imho ok

the cleaner way would be not execute this method elsewhere than in `Host::Managed`, but that would be a big change of existing fact import flows, so I'm ok with this